src/interfaces/widget.ts

```diff
@@ -192,15 +190,30 @@ export interface WidgetTextMeta extends WidgetMeta {
  */
 export type WidgetMetaAny = WidgetFormMeta | WidgetTableMeta | WidgetTextMeta | WidgetInfoMeta
 
+/**
+ * Component of custom widget
+ */
```

Comment:
We probably should deprecate this in favor of `React.ComponentType<any>` and remove in 2.0.0

Reply:
Do you mean change 
```typescript
export type CustomWidget = ConnectedComponent<any, any> | FunctionComponent<any>
```
to
```typescript
export type CustomWidget = ConnectedComponent<any, any> | ComponentType<any>`
```
in case client has class components?

And I haven't got what are you talking about here:
> and remove in 2.0.0

Reply:
I meant:
```ts
/*
 * @deprecated TODO: Remove in 2.0.0
 */
export type CustomWidget = ComponentType<any>
```
The meaning of our `CustomWidget` type is simply "any react component declaration"; `ComponentType<any>` is exactly that as it's an alias for [class/FC union](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0fa935e2b95688bc5e3bcb0290b1b3fd75323dec/types/react/index.d.ts#L78):
```ts
type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```
`ConnectedComponent` is a bit trickier but overlaps enough with `ComponentType` to pass as one:
1) It's an [alias for NamedExoticComponent](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7ef36a683ef8b44ca1e478afed1cdb271a26671/types/react-redux/index.d.ts#L117) with some extensions
2) `NamedExoticComponent` is [an extension of ExoticComponent](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7ef36a683ef8b44ca1e478afed1cdb271a26671/types/react/index.d.ts#L357) with `displayName` prop
3) `ExoticComponent` is an interface for [function type of (props: P) => ReactElement | null](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7ef36a683ef8b44ca1e478afed1cdb271a26671/types/react/index.d.ts#L349) (with typeof) so it will pass as function component which is [the same](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7ef36a683ef8b44ca1e478afed1cdb271a26671/types/react/index.d.ts#L540) with a bunch of optional properties
  [derived](url) from `ComponentType`

Reply:
@Dergash Thank you